src/Aspire.Hosting/Publishing/ManifestPublisher.cs

```diff
@@ -48,68 +48,69 @@ protected virtual async Task PublishInternalAsync(DistributedApplicationModel mo
     protected async Task WriteManifestAsync(DistributedApplicationModel model, Utf8JsonWriter jsonWriter, CancellationToken cancellationToken)
     {
         var manifestPath = _options.Value.OutputPath ?? throw new DistributedApplicationException("The '--output-path [path]' option was not specified even though '--publisher manifest' argument was used.");
-        var context = new ManifestPublishingContext(_executionContext, manifestPath, jsonWriter);
+        var context = new ManifestPublishingContext(_executionContext, manifestPath, jsonWriter, cancellationToken);
 
         jsonWriter.WriteStartObject();
-        WriteResources(model, context);
+        await WriteResources(model, context).ConfigureAwait(false);
         jsonWriter.WriteEndObject();
 
         await jsonWriter.FlushAsync(cancellationToken).ConfigureAwait(false);
     }
 
-    private static void WriteResources(DistributedApplicationModel model, ManifestPublishingContext context)
+    private static async Task WriteResources(DistributedApplicationModel model, ManifestPublishingContext context)
```

Comment:
nit: Add Async suffix to methods where needed

Reply:
Done